enable org.rust.cargo.fetch.actual.stdlib.metadata cause get stdlib failed

<!--
Hello and thank you for the issue!
If you would like to report a bug, we have added some points below that you can fill out.
Consider using "Help | Create New Issue" action that fills out all necessary information automatically.
Feel free to remove all the irrelevant text to request a new feature.
-->

## Environment

* **IntelliJ Rust plugin version:** 0.3.141.3655-203-nightly
* **Rust toolchain version:** rustc 1.51.0-nightly (04caa632d 2021-01-30)
* **IDE name and version:** clion 2020.3.2
* **Operating system:** Archlinux

## Problem description
enable experience feature `org.rust.cargo.fetch.actual.stdlib.metadata` cause intellij-rust get stdlib failed

## Steps to reproduce
1. enable experience feature `org.rust.cargo.fetch.actual.stdlib.metadata`
2. refresh cargo by pressing the refresh button

I run clion on terminal and got some logs, maybe it's helpful

```
2021-02-09 01:01:52,847 [  33892]   WARN - st.cargo.toolchain.tools.Cargo - Execution failed (exit code 101).
/usr/bin/cargo check --message-format json
stdout : {"reason":"compiler-artifact","package_id":"proc-macro2 1.0.24 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.24/build.rs","edition":"2018","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","proc-macro"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/proc-macro2-a5a5780f735c08f5/build-script-build"],"executable":null,"fresh":true}
   Compiling unicode-xid v0.2.1
{"reason":"compiler-artifact","package_id":"syn 1.0.60 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-1.0.60/build.rs","edition":"2018","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["clone-impls","default","derive","full","parsing","printing","proc-macro","quote","visit-mut"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/syn-a68cdf52079f5b05/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"libc 0.2.77 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.77/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["align","default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/libc-5836bce6410f896a/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"memchr 2.3.3 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/memchr-2.3.3/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/memchr-2b0db180473edd54/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"pin-project-internal 0.4.23 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/pin-project-internal-0.4.23/build.rs","edition":"2018","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/pin-project-internal-e03b84dfdfddf1f7/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"cfg-if 0.1.10 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"cfg-if","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/cfg-if-0.1.10/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libcfg_if-4b0a7d1b13381df7.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"once_cell 1.5.2 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"once_cell","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/once_cell-1.5.2/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["alloc","default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libonce_cell-d3ff5d109eb6c70d.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"proc-macro-nested 0.1.6 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro-nested-0.1.6/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/proc-macro-nested-f0de5fb3dbddfec4/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"log 0.4.11 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.11/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/log-3a9ea83eb03ac521/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"cfg-if 1.0.0 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"cfg-if","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/cfg-if-1.0.0/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libcfg_if-d081dc850664d3b7.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"proc-macro-hack 0.5.18 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["proc-macro"],"crate_types":["proc-macro"],"name":"proc-macro-hack","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro-hack-0.5.18/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libproc_macro_hack-f3612094931fdf85.so"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"futures-core 0.3.5 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"futures-core","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.5/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["alloc","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libfutures_core-7256550187103e7b.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"futures-sink 0.3.5 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"futures-sink","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.5/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["alloc","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libfutures_sink-7a6390a1d6a9c47f.rmeta"],"executable":null,"fresh":true}
   Compiling autocfg v1.0.1
{"reason":"compiler-artifact","package_id":"slab 0.4.2 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"slab","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/slab-0.4.2/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libslab-76e40229b1b82562.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"futures-io 0.3.5 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"futures-io","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-io-0.3.5/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libfutures_io-1618fc4a568b5a12.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"smallvec 1.6.1 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"smallvec","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/smallvec-1.6.1/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libsmallvec-88fd51d1a60a27dd.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"scopeguard 1.1.0 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"scopeguard","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/scopeguard-1.1.0/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libscopeguard-2b0c17ff3a9d1632.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"pin-utils 0.1.0 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"pin-utils","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/pin-utils-0.1.0/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libpin_utils-5739094a252a604f.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"bytes 1.0.1 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"bytes","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/bytes-1.0.1/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libbytes-0337cf4c454210d3.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"bytes 0.5.6 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"bytes","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/bytes-0.5.6/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libbytes-cbac4f3e335ed5e3.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"lazy_static 1.4.0 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"lazy_static","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/liblazy_static-88e83b46113e8454.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"fnv 1.0.7 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"fnv","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/fnv-1.0.7/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libfnv-98477584957c8764.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"itoa 0.4.6 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"itoa","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/itoa-0.4.6/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libitoa-c686ccfaa2bbe21b.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"httparse 1.3.4 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/httparse-1.3.4/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/httparse-09661d22663f554a/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"try-lock 0.2.3 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"try-lock","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/try-lock-0.2.3/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libtry_lock-b5da81355a862570.rmeta"],"executable":null,"fresh":true}
error: could not compile `unicode-xid`

Caused by:
  could not execute process `/SSD-speed/sherlock/.local/share/JetBrains/Toolbox/apps/CLion/ch-0/203.7148.70.plugins/intellij-rust/bin/linux/x86-64/intellij-rust-native-helper /usr/bin/rustc --crate-name unicode_xid /home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/unicode-xid-0.2.1/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' -C metadata=083414e3434d8e82 -C extra-filename=-083414e3434d8e82 --out-dir /home/sherlock/Documents/program/rust/learn/target/debug/deps -L dependency=/home/sherlock/Documents/program/rust/learn/target/debug/deps --cap-lints allow -C link-arg=-fuse-ld=lld` (never executed)

Caused by:
  Permission denied (os error 13)
warning: build failed, waiting for other jobs to finish...
{"reason":"build-finished","success":false}
error: build failed

stderr :  
com.intellij.execution.ExecutionException: Execution failed (exit code 101).
/usr/bin/cargo check --message-format json
stdout : {"reason":"compiler-artifact","package_id":"proc-macro2 1.0.24 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.24/build.rs","edition":"2018","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","proc-macro"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/proc-macro2-a5a5780f735c08f5/build-script-build"],"executable":null,"fresh":true}
   Compiling unicode-xid v0.2.1
{"reason":"compiler-artifact","package_id":"syn 1.0.60 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-1.0.60/build.rs","edition":"2018","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["clone-impls","default","derive","full","parsing","printing","proc-macro","quote","visit-mut"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/syn-a68cdf52079f5b05/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"libc 0.2.77 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.77/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["align","default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/libc-5836bce6410f896a/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"memchr 2.3.3 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/memchr-2.3.3/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/memchr-2b0db180473edd54/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"pin-project-internal 0.4.23 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/pin-project-internal-0.4.23/build.rs","edition":"2018","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/pin-project-internal-e03b84dfdfddf1f7/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"cfg-if 0.1.10 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"cfg-if","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/cfg-if-0.1.10/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libcfg_if-4b0a7d1b13381df7.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"once_cell 1.5.2 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"once_cell","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/once_cell-1.5.2/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["alloc","default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libonce_cell-d3ff5d109eb6c70d.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"proc-macro-nested 0.1.6 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro-nested-0.1.6/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/proc-macro-nested-f0de5fb3dbddfec4/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"log 0.4.11 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.11/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/log-3a9ea83eb03ac521/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"cfg-if 1.0.0 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"cfg-if","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/cfg-if-1.0.0/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libcfg_if-d081dc850664d3b7.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"proc-macro-hack 0.5.18 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["proc-macro"],"crate_types":["proc-macro"],"name":"proc-macro-hack","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro-hack-0.5.18/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libproc_macro_hack-f3612094931fdf85.so"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"futures-core 0.3.5 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"futures-core","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.5/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["alloc","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libfutures_core-7256550187103e7b.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"futures-sink 0.3.5 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"futures-sink","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-sink-0.3.5/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["alloc","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libfutures_sink-7a6390a1d6a9c47f.rmeta"],"executable":null,"fresh":true}
   Compiling autocfg v1.0.1
{"reason":"compiler-artifact","package_id":"slab 0.4.2 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"slab","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/slab-0.4.2/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libslab-76e40229b1b82562.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"futures-io 0.3.5 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"futures-io","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-io-0.3.5/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libfutures_io-1618fc4a568b5a12.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"smallvec 1.6.1 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"smallvec","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/smallvec-1.6.1/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libsmallvec-88fd51d1a60a27dd.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"scopeguard 1.1.0 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"scopeguard","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/scopeguard-1.1.0/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libscopeguard-2b0c17ff3a9d1632.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"pin-utils 0.1.0 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"pin-utils","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/pin-utils-0.1.0/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libpin_utils-5739094a252a604f.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"bytes 1.0.1 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"bytes","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/bytes-1.0.1/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libbytes-0337cf4c454210d3.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"bytes 0.5.6 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"bytes","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/bytes-0.5.6/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libbytes-cbac4f3e335ed5e3.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"lazy_static 1.4.0 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"lazy_static","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/liblazy_static-88e83b46113e8454.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"fnv 1.0.7 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"fnv","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/fnv-1.0.7/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libfnv-98477584957c8764.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"itoa 0.4.6 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"itoa","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/itoa-0.4.6/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libitoa-c686ccfaa2bbe21b.rmeta"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"httparse 1.3.4 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["custom-build"],"crate_types":["bin"],"name":"build-script-build","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/httparse-1.3.4/build.rs","edition":"2015","doc":false,"doctest":false,"test":false},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":["default","std"],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/build/httparse-09661d22663f554a/build-script-build"],"executable":null,"fresh":true}
{"reason":"compiler-artifact","package_id":"try-lock 0.2.3 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"try-lock","src_path":"/home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/try-lock-0.2.3/src/lib.rs","edition":"2015","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"0","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":false},"features":[],"filenames":["/home/sherlock/Documents/program/rust/learn/target/debug/deps/libtry_lock-b5da81355a862570.rmeta"],"executable":null,"fresh":true}
error: could not compile `unicode-xid`

Caused by:
  could not execute process `/SSD-speed/sherlock/.local/share/JetBrains/Toolbox/apps/CLion/ch-0/203.7148.70.plugins/intellij-rust/bin/linux/x86-64/intellij-rust-native-helper /usr/bin/rustc --crate-name unicode_xid /home/sherlock/.cargo/registry/src/github.com-1ecc6299db9ec823/unicode-xid-0.2.1/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' -C metadata=083414e3434d8e82 -C extra-filename=-083414e3434d8e82 --out-dir /home/sherlock/Documents/program/rust/learn/target/debug/deps -L dependency=/home/sherlock/Documents/program/rust/learn/target/debug/deps --cap-lints allow -C link-arg=-fuse-ld=lld` (never executed)

Caused by:
  Permission denied (os error 13)
warning: build failed, waiting for other jobs to finish...
{"reason":"build-finished","success":false}
error: build failed

stderr : 
	at org.rust.openapiext.CommandLineExtKt.execute(CommandLineExt.kt:93)
	at org.rust.cargo.toolchain.tools.Cargo.execute(Cargo.kt:416)
	at org.rust.cargo.toolchain.tools.Cargo.execute$default(Cargo.kt:415)
	at org.rust.cargo.toolchain.tools.Cargo.fetchBuildScriptsInfo(Cargo.kt:186)
	at org.rust.cargo.toolchain.tools.Cargo.fullProjectDescription(Cargo.kt:129)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt$fetchCargoWorkspace$1.invoke(CargoSyncTask.kt:213)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt$fetchCargoWorkspace$1.invoke(CargoSyncTask.kt)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress(CargoSyncTask.kt:290)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$runWithChildProgress(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTask$SyncContext.runWithChildProgress(CargoSyncTask.kt:168)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.fetchCargoWorkspace(CargoSyncTask.kt:203)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$fetchCargoWorkspace(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTask$doRun$$inlined$map$lambda$1.invoke(CargoSyncTask.kt:112)
	at org.rust.cargo.project.model.impl.CargoSyncTask$doRun$$inlined$map$lambda$1.invoke(CargoSyncTask.kt:50)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress(CargoSyncTask.kt:290)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress$default(CargoSyncTask.kt:285)
	at org.rust.cargo.project.model.impl.CargoSyncTask.doRun(CargoSyncTask.kt:100)
	at org.rust.cargo.project.model.impl.CargoSyncTask.run(CargoSyncTask.kt:70)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:962)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsync$5(CoreProgressManager.java:472)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:235)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:65)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:235)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
2021-02-09 01:02:36,831 [  77876]  ERROR -    #org.rust.stdext.AsyncValue - java.lang.IllegalStateException: `cargo metadata` reported a package which does not exist at `/home/sherlock/.cache/JetBrains/CLion2020.3/intellij-rust/stdlib/1.51.0-nightly-47c4bc9ffa64220b7d39445d19f81f75b31283ed/vendor/rustc-std-workspace-core/Cargo.toml` 
java.util.concurrent.CompletionException: java.lang.IllegalStateException: `cargo metadata` reported a package which does not exist at `/home/sherlock/.cache/JetBrains/CLion2020.3/intellij-rust/stdlib/1.51.0-nightly-47c4bc9ffa64220b7d39445d19f81f75b31283ed/vendor/rustc-std-workspace-core/Cargo.toml`
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:632)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at org.rust.cargo.project.model.impl.CargoSyncTask.run(CargoSyncTask.kt:84)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:962)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsync$5(CoreProgressManager.java:472)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:235)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:65)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:235)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: `cargo metadata` reported a package which does not exist at `/home/sherlock/.cache/JetBrains/CLion2020.3/intellij-rust/stdlib/1.51.0-nightly-47c4bc9ffa64220b7d39445d19f81f75b31283ed/vendor/rustc-std-workspace-core/Cargo.toml`
	at org.rust.cargo.toolchain.impl.CargoMetadata.clean(CargoMetadata.kt:369)
	at org.rust.cargo.toolchain.impl.CargoMetadata.clean(CargoMetadata.kt:341)
	at org.rust.cargo.toolchain.impl.CargoMetadata.clean$default(CargoMetadata.kt:326)
	at org.rust.cargo.project.workspace.StdlibDataFetcher.fetchStdlibData(StandardLibrary.kt:185)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fetchActualStdlib(StandardLibrary.kt:88)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fromFile(StandardLibrary.kt:78)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fromFile$default(StandardLibrary.kt:59)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.fetchStdlib(CargoSyncTask.kt:270)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$fetchStdlib(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt$fetchStdlib$1.invoke(CargoSyncTask.kt:262)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt$fetchStdlib$1.invoke(CargoSyncTask.kt)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress(CargoSyncTask.kt:290)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$runWithChildProgress(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTask$SyncContext.runWithChildProgress(CargoSyncTask.kt:168)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.fetchStdlib(CargoSyncTask.kt:234)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$fetchStdlib(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTask$doRun$$inlined$map$lambda$1.invoke(CargoSyncTask.kt:113)
	at org.rust.cargo.project.model.impl.CargoSyncTask$doRun$$inlined$map$lambda$1.invoke(CargoSyncTask.kt:50)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress(CargoSyncTask.kt:290)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress$default(CargoSyncTask.kt:285)
	at org.rust.cargo.project.model.impl.CargoSyncTask.doRun(CargoSyncTask.kt:100)
	at org.rust.cargo.project.model.impl.CargoSyncTask.run(CargoSyncTask.kt:70)
	... 17 more
2021-02-09 01:02:36,836 [  77881]  ERROR -    #org.rust.stdext.AsyncValue - CLion 2020.3.2  Build #CL-203.7148.70 
2021-02-09 01:02:36,836 [  77881]  ERROR -    #org.rust.stdext.AsyncValue - JDK: 11.0.9.1; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2021-02-09 01:02:36,836 [  77881]  ERROR -    #org.rust.stdext.AsyncValue - OS: Linux 
2021-02-09 01:02:36,836 [  77881]  ERROR -    #org.rust.stdext.AsyncValue - Plugin to blame: Rust version: 0.3.141.3655-203-nightly 
2021-02-09 01:02:36,836 [  77881]  ERROR -    #org.rust.stdext.AsyncValue - Last Action: Cargo.RefreshCargoProject 
2021-02-09 01:02:36,838 [  77883]  ERROR - intellij.openapi.progress.Task - `cargo metadata` reported a package which does not exist at `/home/sherlock/.cache/JetBrains/CLion2020.3/intellij-rust/stdlib/1.51.0-nightly-47c4bc9ffa64220b7d39445d19f81f75b31283ed/vendor/rustc-std-workspace-core/Cargo.toml` 
java.lang.IllegalStateException: `cargo metadata` reported a package which does not exist at `/home/sherlock/.cache/JetBrains/CLion2020.3/intellij-rust/stdlib/1.51.0-nightly-47c4bc9ffa64220b7d39445d19f81f75b31283ed/vendor/rustc-std-workspace-core/Cargo.toml`
	at org.rust.cargo.toolchain.impl.CargoMetadata.clean(CargoMetadata.kt:369)
	at org.rust.cargo.toolchain.impl.CargoMetadata.clean(CargoMetadata.kt:341)
	at org.rust.cargo.toolchain.impl.CargoMetadata.clean$default(CargoMetadata.kt:326)
	at org.rust.cargo.project.workspace.StdlibDataFetcher.fetchStdlibData(StandardLibrary.kt:185)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fetchActualStdlib(StandardLibrary.kt:88)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fromFile(StandardLibrary.kt:78)
	at org.rust.cargo.project.workspace.StandardLibrary$Companion.fromFile$default(StandardLibrary.kt:59)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.fetchStdlib(CargoSyncTask.kt:270)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$fetchStdlib(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt$fetchStdlib$1.invoke(CargoSyncTask.kt:262)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt$fetchStdlib$1.invoke(CargoSyncTask.kt)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress(CargoSyncTask.kt:290)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$runWithChildProgress(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTask$SyncContext.runWithChildProgress(CargoSyncTask.kt:168)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.fetchStdlib(CargoSyncTask.kt:234)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.access$fetchStdlib(CargoSyncTask.kt:1)
	at org.rust.cargo.project.model.impl.CargoSyncTask$doRun$$inlined$map$lambda$1.invoke(CargoSyncTask.kt:113)
	at org.rust.cargo.project.model.impl.CargoSyncTask$doRun$$inlined$map$lambda$1.invoke(CargoSyncTask.kt:50)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress(CargoSyncTask.kt:290)
	at org.rust.cargo.project.model.impl.CargoSyncTaskKt.runWithChildProgress$default(CargoSyncTask.kt:285)
	at org.rust.cargo.project.model.impl.CargoSyncTask.doRun(CargoSyncTask.kt:100)
	at org.rust.cargo.project.model.impl.CargoSyncTask.run(CargoSyncTask.kt:70)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:962)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsync$5(CoreProgressManager.java:472)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:235)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:65)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:235)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
2021-02-09 01:02:36,839 [  77884]  ERROR - intellij.openapi.progress.Task - CLion 2020.3.2  Build #CL-203.7148.70 
2021-02-09 01:02:36,839 [  77884]  ERROR - intellij.openapi.progress.Task - JDK: 11.0.9.1; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2021-02-09 01:02:36,839 [  77884]  ERROR - intellij.openapi.progress.Task - OS: Linux 
2021-02-09 01:02:36,839 [  77884]  ERROR - intellij.openapi.progress.Task - Plugin to blame: Rust version: 0.3.141.3655-203-nightly 
2021-02-09 01:02:36,839 [  77884]  ERROR - intellij.openapi.progress.Task - Last Action: Cargo.RefreshCargoProject 
2021-02-09 01:04:57,037 [ 218082]   WARN - tions.ex.ConfigurableCardPanel - auto-dispose 'Plugins' id=preferences.pluginManager 
2021-02-09 01:05:02,201 [ 223246]   WARN - com.intellij.ide.IdeEventQueue - Too many mouse clicks (15)!!! 
2021-02-09 01:05:02,211 [ 223256]   WARN - com.intellij.ide.IdeEventQueue - Too many mouse clicks (15)!!! 
2021-02-09 01:05:26,247 [ 247292]   WARN - com.intellij.util.xmlb.Binding - no accessors for org.jetbrains.idea.perforce.perforce.ConnectionId
```

--- a/src/main/kotlin/org/rust/cargo/project/workspace/StandardLibrary.kt
+++ b/src/main/kotlin/org/rust/cargo/project/workspace/StandardLibrary.kt
@@ -279,9 +279,6 @@ private class StdlibDataFetcher private constructor(
             testPackageSrcDir: VirtualFile,
             version: RustcVersion
         ): Path? {
-            val stdlibDependenciesDir = srcDir.pathAsPath.resolve("../vendor").normalize()
-            if (stdlibDependenciesDir.exists()) return stdlibDependenciesDir
-
             val stdlibHash = stdlibHash(srcDir, version)
 
             val stdlibVendor = RsPathManager.stdlibDependenciesDir().resolve("${version.semver.parsedVersion}-$stdlibHash/vendor")

--- a/src/main/kotlin/org/rust/cargo/project/workspace/StandardLibrary.kt
+++ b/src/main/kotlin/org/rust/cargo/project/workspace/StandardLibrary.kt
@@ -5,18 +5,22 @@
 
 package org.rust.cargo.project.workspace
 
+import com.google.common.annotations.VisibleForTesting
 import com.intellij.execution.ExecutionException
 import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.diagnostic.logger
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.LocalFileSystem
 import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapiext.isUnitTestMode
+import com.intellij.util.io.delete
 import com.intellij.util.io.exists
 import org.rust.cargo.CargoConstants
 import org.rust.cargo.CfgOptions
 import org.rust.cargo.project.model.RustcInfo
 import org.rust.cargo.project.settings.toolchain
 import org.rust.cargo.toolchain.impl.CargoMetadata
+import org.rust.cargo.toolchain.impl.CargoMetadataException
 import org.rust.cargo.toolchain.impl.RustcVersion
 import org.rust.cargo.toolchain.tools.Cargo
 import org.rust.cargo.toolchain.tools.cargo
@@ -28,6 +32,7 @@ import org.rust.openapiext.isFeatureEnabled
 import org.rust.openapiext.pathAsPath
 import org.rust.stdext.HashCode
 import org.rust.stdext.toPath
+import java.io.IOException
 import java.nio.file.Path
 
 data class StandardLibrary(
@@ -58,13 +63,7 @@ data class StandardLibrary(
             rustcInfo: RustcInfo?,
             isPartOfCargoProject: Boolean = false
         ): StandardLibrary? {
-            if (!sources.isDirectory) return null
-
-            val srcDir = if (sources.name in SRC_ROOTS) {
-                sources
-            } else {
-                sources.findFirstFileByRelativePaths(SRC_ROOTS) ?: sources
-            }
+            val srcDir = findSrcDir(sources) ?: return null
 
             val stdlib = if (isFeatureEnabled(RsExperiments.FETCH_ACTUAL_STDLIB_METADATA) && !isPartOfCargoProject) {
                 val rustcVersion = rustcInfo?.version
@@ -73,7 +72,9 @@ data class StandardLibrary(
                     LOG.warn("Toolchain version is unknown. Hardcoded stdlib struture will be used")
                     fetchHardcodedStdlib(srcDir)
                 } else {
-                    fetchActualStdlib(project, srcDir, rustcVersion)
+                    val result = fetchActualStdlib(project, srcDir, rustcVersion)
+                    // TODO: show some warning to users if result is null
+                    result ?: fetchHardcodedStdlib(srcDir)
                 }
             } else {
                 fetchHardcodedStdlib(srcDir)
@@ -82,8 +83,34 @@ data class StandardLibrary(
             return stdlib?.copy(isPartOfCargoProject = isPartOfCargoProject)
         }
 
-        private fun fetchActualStdlib(project: Project, srcDir: VirtualFile, version: RustcVersion): StandardLibrary? {
-            return StdlibDataFetcher.create(project, srcDir, version)?.fetchStdlibData()
+        @VisibleForTesting
+        fun findSrcDir(sources: VirtualFile): VirtualFile? {
+            if (!sources.isDirectory) return null
+            return if (sources.name in SRC_ROOTS) {
+                sources
+            } else {
+                sources.findFirstFileByRelativePaths(SRC_ROOTS) ?: sources
+            }
+        }
+
+        private fun fetchActualStdlib(
+            project: Project,
+            srcDir: VirtualFile,
+            version: RustcVersion,
+            cleanVendorDir: Boolean = false
+        ): StandardLibrary? {
+            try {
+                return StdlibDataFetcher.create(project, srcDir, version, cleanVendorDir)?.fetchStdlibData()
+            } catch (e: Throwable) {
+                if (!isUnitTestMode) {
+                    // Logger.error in tests, fail the test
+                    LOG.error(e)
+                }
+                if (!cleanVendorDir && e is CargoMetadataException) {
+                    return fetchActualStdlib(project, srcDir, version, cleanVendorDir = true)
+                }
+            }
+            return null
         }
 
         private fun fetchHardcodedStdlib(srcDir: VirtualFile): StandardLibrary? {
@@ -142,7 +169,7 @@ data class StandardLibrary(
     }
 }
 
-private class StdlibDataFetcher private constructor(
+class StdlibDataFetcher private constructor(
     private val project: Project,
     private val cargo: Cargo,
     private val srcDir: VirtualFile,
@@ -261,15 +288,28 @@ private class StdlibDataFetcher private constructor(
     companion object {
         private val LOG: Logger = logger<StdlibDataFetcher>()
 
-        fun create(project: Project, srcDir: VirtualFile, version: RustcVersion): StdlibDataFetcher? {
+        fun create(project: Project, srcDir: VirtualFile, version: RustcVersion, cleanVendorDir: Boolean): StdlibDataFetcher? {
             val cargo = project.toolchain?.cargo() ?: return null
 
             val testPackageSrcPaths = listOf(AutoInjectedCrates.TEST, "lib${AutoInjectedCrates.TEST}")
             val testPackageSrcDir = srcDir.findFirstFileByRelativePaths(testPackageSrcPaths)?.canonicalFile
                 ?: return null
-            val stdlibDependenciesDir = findStdlibDependencyDirectory(project, cargo, srcDir, testPackageSrcDir, version)
+            val stdlibDependenciesDir = findStdlibDependencyDirectory(
+                project,
+                cargo,
+                srcDir,
+                testPackageSrcDir,
+                version,
+                cleanVendorDir
+            )
                 ?: return null
             return StdlibDataFetcher(project, cargo, srcDir, testPackageSrcDir, stdlibDependenciesDir)
+        }
+
+        @VisibleForTesting
+        fun stdlibVendorDir(srcDir: VirtualFile, version: RustcVersion): Path {
+            val stdlibHash = stdlibHash(srcDir, version)
+            return RsPathManager.stdlibDependenciesDir().resolve("${version.semver.parsedVersion}-$stdlibHash/vendor")
         }
 
         private fun findStdlibDependencyDirectory(
@@ -277,12 +317,22 @@ private class StdlibDataFetcher private constructor(
             cargo: Cargo,
             srcDir: VirtualFile,
             testPackageSrcDir: VirtualFile,
-            version: RustcVersion
+            version: RustcVersion,
+            cleanVendorDir: Boolean
         ): Path? {
-            val stdlibHash = stdlibHash(srcDir, version)
+            val stdlibVendor = stdlibVendorDir(srcDir, version)
+            var stdlibVendorExists = stdlibVendor.exists()
+            if (stdlibVendorExists && cleanVendorDir) {
+                try {
+                    stdlibVendor.delete(recursively = true)
+                } catch (e: IOException) {
+                    LOG.error(e)
+                    return null
+                }
+                stdlibVendorExists = false
+            }
 
-            val stdlibVendor = RsPathManager.stdlibDependenciesDir().resolve("${version.semver.parsedVersion}-$stdlibHash/vendor")
-            if (!stdlibVendor.exists()) {
+            if (!stdlibVendorExists) {
                 try {
                     // `test` package depends on all other stdlib packages,
                     // so it's enough to vendor only its dependencies

--- a/src/main/kotlin/org/rust/cargo/toolchain/impl/CargoMetadata.kt
+++ b/src/main/kotlin/org/rust/cargo/toolchain/impl/CargoMetadata.kt
@@ -360,7 +360,7 @@ object CargoMetadata {
         val rootPath = PathUtil.getParentPath(manifest_path)
         val root = fs.refreshAndFindFileByPath(rootPath)
             ?.let { if (isWorkspaceMember) it else it.canonicalFile }
-        checkNotNull(root) { "`cargo metadata` reported a package which does not exist at `$manifest_path`" }
+            ?: throw CargoMetadataException("`cargo metadata` reported a package which does not exist at `$manifest_path`")
 
         val features = features.toMutableMap()
 

--- a/src/main/kotlin/org/rust/cargo/toolchain/impl/CargoMetadataException.kt
+++ b/src/main/kotlin/org/rust/cargo/toolchain/impl/CargoMetadataException.kt
@@ -1,0 +1,8 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.toolchain.impl
+
+class CargoMetadataException(message: String) : IllegalStateException(message)

--- a/src/test/kotlin/org/rust/cargo/project/model/CargoStdlibPackagesTest.kt
+++ b/src/test/kotlin/org/rust/cargo/project/model/CargoStdlibPackagesTest.kt
@@ -5,45 +5,79 @@
 
 package org.rust.cargo.project.model
 
-import org.rust.MinRustcVersion
+import com.intellij.util.io.delete
 import org.rust.cargo.RsWithToolchainTestBase
-import org.rust.cargo.project.workspace.CargoWorkspace
-import org.rust.cargo.project.workspace.FeatureState
-import org.rust.cargo.project.workspace.PackageOrigin
-import org.rust.ide.experiments.RsExperiments
-import org.rust.openapiext.runWithEnabledFeatures
+import org.rust.cargo.project.model.impl.testCargoProjects
+import org.rust.cargo.project.workspace.*
 import org.rust.singleProject
 import org.rust.workspaceOrFail
 
-@MinRustcVersion("1.41.0")
 class CargoStdlibPackagesTest : RsWithToolchainTestBase() {
 
-    fun `test stdlib dependency`() {
-        runWithEnabledFeatures(RsExperiments.FETCH_ACTUAL_STDLIB_METADATA) {
-            buildProject {
-                toml("Cargo.toml", """
-                    [package]
-                    name = "sandbox"
-                    version = "0.1.0"
-                    authors = []
-                """)
+    override val fetchActualStdlibMetadata: Boolean get() = true
 
-                dir("src") {
-                    rust("lib.rs", "")
-                }
+    fun `test stdlib dependency`() {
+        buildProject {
+            toml("Cargo.toml", """
+                [package]
+                name = "sandbox"
+                version = "0.1.0"
+                authors = []
+            """)
+
+            dir("src") {
+                rust("lib.rs", "")
             }
         }
 
         val cargoProject = project.cargoProjects.singleProject()
         val workspace = cargoProject.workspaceOrFail()
-        val hashbrownPkg = workspace.packages.first { it.name == "hashbrown" }
+        val hashbrownPkg = workspace.packages.first { it.name == HASHBROWN }
         assertEquals(PackageOrigin.STDLIB_DEPENDENCY, hashbrownPkg.origin)
         hashbrownPkg.checkFeature("rustc-dep-of-std", FeatureState.Enabled)
         hashbrownPkg.checkFeature("default", FeatureState.Disabled)
     }
 
+    fun `test recover corrupted stdlib dependency directory`() {
+        buildProject {
+            toml("Cargo.toml", """
+                [package]
+                name = "sandbox"
+                version = "0.1.0"
+                authors = []
+            """)
+
+            dir("src") {
+                rust("lib.rs", "")
+            }
+        }
+
+        val cargoProjectService = project.testCargoProjects
+        val cargoProject = cargoProjectService.singleProject()
+        val version = cargoProject.rustcInfo?.version ?: error("")
+        val srcDir = rustupFixture.stdlib?.let { StandardLibrary.findSrcDir(it) } ?: error("")
+
+        val path = StdlibDataFetcher.stdlibVendorDir(srcDir, version)
+        // Corrupt vendor directory
+        path.resolve(HASHBROWN).delete(true)
+
+        cargoProjectService.refreshAllProjectsSync()
+
+        val hashbrownPkg = cargoProjectService
+            .singleProject()
+            .workspaceOrFail()
+            .packages
+            .firstOrNull { it.name == HASHBROWN }
+        assertNotNull("Stdlib must contain `$HASHBROWN` package", hashbrownPkg)
+
+    }
+
     private fun CargoWorkspace.Package.checkFeature(featureName: String, expectedState: FeatureState) {
         val featureState = featureState.getValue(featureName)
-        assertEquals("Feature `$featureName` in pakcage `$name` should be in $expectedState", expectedState, featureState)
+        assertEquals("Feature `$featureName` in package `$name` should be in $expectedState", expectedState, featureState)
+    }
+
+    companion object {
+        private const val HASHBROWN = "hashbrown"
     }
 }

